### ISSUE TITLE
[Enhancement] Stable sort for elements in sequences.

### ISSUE THREAD

**Issue by Malhavok** [Demigiant/dotween#131]
TL;DR:
Consider changing List.Sort in Sequence.cs:169 to Linq.OrderBy.

Long story:
I know, at first this change seems to be kinda... bad. The difference between those two is that List.Sort is unstable (doesn't preserve order of the same value elements) while Linq.OrderBy is.

What this gives, is that it ensures that callbacks Append'ed to Sequence will be last in order thus called after previous tween ends. In some rare cases on some weird platforms (so far I've only found some issues on Linux and WebGL and only in some specific cases where callback calls GameObject.Destroy on target) order of execution does matter. Simple solution is to add some small epsilon interval (I tend to use 1e-4f) before callback. It would be nice though, if "Append" would really append.

**Comment by Demigiant** [comment]
Hey thanks for the tip, will check it out. I don't want to add any Linq references to DOTween's library, but I will check if I can make a custom method. I'm afraid though that the reason Sequences at the same time don't happen in the correct order is not just related to the sorting.

**Comment by Malhavok** [comment]
I might be able to prepare minimal test at the weekend, showing some randomly happening problems with long/complicated sequences.
The main issue is, that it doesn't happen in the editor, ever (I had editor running for 4 days constantly without single occurance of an error, while on target linux the error happens in less than 5 hours...).
I also never noticed any errors on macOS, iOS nor Android targets. Only on Linux and WebGL.

On WebGL I tracked it to null target (after GameObject.Destroy that happens on callback appended to the end of the sequence) being passed to the tween that shouldn't be running any more... That's like a whole frame difference in timing... And it happens like once per 100 Sequences created (without pooling). In SafeMode it just stops Sequence execution, without SafeMode it throws an exception and works properly.

Maybe something changed in IL2CPP, and that breaks something?

Sidenote: I've tested this issue with each DOTween version from 1.1.340 up.
